client/src/components/HistoryViewer/HistoryViewerVersion.js

```diff
@@ -178,50 +177,46 @@ class HistoryViewerVersion extends Component {
 }
 
 HistoryViewerVersion.propTypes = {
+  version: versionType,
   isActive: React.PropTypes.bool,
   onSelect: React.PropTypes.func,
-  onCompare: React.PropTypes.func,
-  onCompareSelect: React.PropTypes.func,
+  enterCompareMode: React.PropTypes.func,
   compare: compareType,
   StateComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
   FormActionComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
-  version: versionType,
 };
 
 HistoryViewerVersion.defaultProps = {
   isActive: false,
   version: defaultVersion,
 };
 
-function mapStateToProps(state) {
-  return {
-    compare: state.versionedAdmin.historyViewer.compare,
-  };
-}
+export { HistoryViewerVersion as Component };
 
 function mapDispatchToProps(dispatch) {
   return {
-    onSelect(id) {
-      dispatch(showVersion(id));
-    },
-    onCompare(id) {
-        dispatch(setCompareFrom(id));
-        dispatch(setCompareMode(true));
-    },
-    onCompareSelect(versionID, setFrom) {
-      if (setFrom) {
-        dispatch(setCompareFrom(versionID));
+    onSelect(selectedVersion, compare) {
+      const { versionFrom } = compare;
+      if (compare) {
+        if (!versionFrom) {
+          dispatch(setCompareFrom(selectedVersion));
+        } else {
+          dispatch(setCompareTo(selectedVersion));
+        }
       } else {
-        dispatch(setCompareTo(versionID));
+        dispatch(showVersion(selectedVersion));
+        dispatch(clearMessages());
       }
     },
+    enterCompareMode(version) {
```

Comment:
Can you please rename this to `onEnterCompareMode` or something like that, that uses the `on` prefix?

Reply:
Sure :)